### ISSUE TITLE
Named sockets appear in committed images as empty files

### ISSUE THREAD

**Issue by kingdonb** [docker/docker#4336]
I have this image that is generated from a barely modified https://github.com/phusion/baseimage-docker (updated to work on 13.10, and only to include my public keys) and it works really well for me, it starts up syslog-ng, sshd, cron, until I try to stop one of the running containers and commit it to a new image.

This series of pastebin links with some annotation will explain:
1) the error
`Error binding socket; addr='AF_UNIX(/dev/log)', error='Address already in use (98)'`
in context:
http://pastebin.com/dfUgVXFb

2) nothing seems amiss in docker inspect (second image was committed from a stopped container of the first... the image output by the dockerfile is the "clean" working version, `192.168.3.197:5000/baseimage:saucy`)
http://pastebin.com/iqteXU5B

3) 0a6 is a container from that image generated by the dockerfile, everything working great, a little noise on the proc info for syslog-ng, this is an expected warning and just some noise, you can see syslog starts and reporting proceeds properly, the process does not terminate abnormally...
http://pastebin.com/t4FN6t3U

4) 0a6 can be stopped and started and this does not change anything... all OK still
http://pastebin.com/ve2eGguZ

5) Commit 0a6 to an image called "saucy-broken" where you can see syslog-ng repeatedly fails to start because of /dev/log "already in use", so shutdown the service (`sv d syslog-ng`), it stops eating up pids, and try running it manually to see what fails;
http://pastebin.com/GMSYXEky
These are the steps to reproduce the error from (1).

At first I thought it was a problem in Ubuntu but now I can see, all you have to do to reproduce this error is to try to commit the container manually and run it again with the same parameters.  I am really confused.  Not sure what may be wrong for `/dev/log`.

**Comment by jfrazelle** [comment]
@kingdonb can you let us know if this is still an issue with the latest version of docker

**Comment by kingdonb** [comment]
This was addressed "downstream" in baseimage-docker, but I can revert the fix and check docker's behavior, if you have any reason to believe the situation might have changed.

The solution we came up with was to just know what apps (syslog-ng) like to leave around their named sockets on the filesystem, keep an inventory of those named sockets and their locations, and clear them from the image at startup time if they are ever found to be present as anything other than named sockets.

Obviously not optimal but this seems to be a rare edge-case that is not triggered by a lot of software.
